controllers/course_memberships_controller.js

```diff
@@ -0,0 +1,58 @@
+const db = require('../db');
+const sql = require('../sql');
+const log = require('../helpers/logging');
+const { coursePath } = require('../routes/helpers/courses');
+const { courseMembershipsPath } = require('../routes/helpers/course_memberships');
+
+exports.index = (req, res, next) => {
+  const { semester_name } = req.params;
+  const { module_code } = req.params;
+  db.query(sql.course_memberships.queries.get_memberships, [semester_name, module_code], (err, data) => {
+    if (err) {
+      log.error('Failed to get course memberships');
+      next(err);
+    } else {
+      res.render('courseMemberships', { data: data.rows, semester_name, module_code });
+    }
+  });
+};
+
+exports.new = (req, res) => {
+  const { semester_name } = req.params;
+  const { module_code } = req.params;
+  res.render('courseMembershipNew', { semester_name, module_code });
+};
+
+exports.create = (req, res) => {
+  const { role } = req.body;
+  const { semester_name } = req.body;
+  const { module_code } = req.body;
+  const { user_id } = req.body;
+
+  db.query(sql.course_memberships.queries.create_membership, [role, semester_name, module_code, user_id], err => {
+    if (err) {
+      log.error('Cannot create course membership!');
+      // TODO: refine error message
+      req.flash('error', err.message);
+    } else {
+      req.flash('success', 'Course membership successfully created!');
+    }
+    return res.redirect(coursePath(semester_name, module_code));
+  });
+};
+
+exports.delete = (req, res) => {
+  const { semester_name } = req.params;
+  const { module_code } = req.params;
+  const { user_id } = req.params;
+
+  db.query(sql.course_memberships.queries.delete_membership, [semester_name, module_code, user_id], err => {
+    if (err) {
+      log.error('Failed to delete course membership');
+      req.flash('error', err.message);
+    } else {
+      req.flash('success', 'Successfully deleted course membership');
```

Comment:
Please refine the message to `Successfully deleted (user_id) from course (module_code)!`